src/components/posts/PostProvider.js

```diff
@@ -60,7 +60,7 @@ export const PostProvider = (props) => {
             },
             body: JSON.stringify(post)
         })
-            .then(getPosts)
+            .then(res => res.json())
```

Comment:
This should not have been changed.  You just need to update the posts using `getPosts`.  You don't need to change anything to JSON because the update doesn't respond with any data in the response.

Reply:
This is inside the adding a new post and not updating the post, in adding a post the POST request comes back with created post, and I am using the id of of the newly created post. After that when you go to all posts page it renders the page again to get all the posts with the one newly created one. all the functionality worked when I tested on my side.

I did not change anything inside updating a post because I already have the postId to go to.

Reply:
Oh perfect.  For some reason it was showing differently on my side... that works!